SQL + Hibernate Guide don't depend on Micronaut Data Hibernate

sdelamo opened the pull request:
See: https://github.com/micronaut-projects/micronaut-guides/issues/1356

sdelamo committed:
remove yaml configuration from hibernate JPA guide

See: https://github.com/micronaut-projects/micronaut-guides/issues/1356

properties

properties

sdelamo committed:
replace @ReadOnly with @Transactional

sdelamo committed:
split data source configuration and dependencies

sdelamo committed:
don’t depend on Micronaut Data JPA

timyates reviewed (approved):
I sometimes wonder if we should just move micronaut-sql into micronaut-data as you can't seem to use one without the other 🤔

sdelamo commented:
> I sometimes wonder if we should just move micronaut-sql into micronaut-data as you can't seem to use one without the other 🤔

Yes, there is a weird cycle 

I reported an issue: 

https://github.com/micronaut-projects/micronaut-sql/issues/1157

sdelamo committed:
Update to 4.2.0 (#1370)

* 4.2.0-SNAPSHOT

* don’t specify all-open version

* oracle autonomous db

* Replace MySQL feature to use old dependency

Without this, the Maven/TestResources integration doesn't work as it's using an old version that doesn't know about connector-j

* Delete buildSrc/src/main/java/io/micronaut/guides/feature/FlywayReplacement.java

* Delete buildSrc/src/main/java/io/micronaut/guides/feature/database/MySQLReplacement.java

---------

Co-authored-by: Tim Yates <tim.yates@gmail.com>

sdelamo committed:
Delete buildSrc/src/main/java/io/micronaut/guides/feature/HibernateJpaReplacement.java

sdelamo commented on guides/micronaut-jpa-hibernate/micronaut-jpa-hibernate.adoc:
```diff
@@ -8,38 +8,40 @@ common:completesolution.adoc[]
+dependency:micronaut-data-tx-hibernate[groupId=io.micronaut.data,callout=2]
+dependency:micronaut-jdbc-hikari[groupId=io.micronaut.sql,callout=3]
+dependency:h2[groupId=com.h2database,scope=runtimeOnly,callout=4]
+dependency:micronaut-data-model[groupId=io.micronaut.data]
```
```suggestion
```

sdelamo commented on guides/micronaut-jpa-hibernate/micronaut-jpa-hibernate.adoc:
```diff
@@ -8,38 +8,40 @@ common:completesolution.adoc[]
+<2> Adds Micronaut Data Transaction Hibernate dependency.
+<3> Configures SQL DataSource instances using Hikari Connection Pool.
+<4> Add dependency to in-memory H2 Database.
+<5> Add dependency to Jakarta Persistence API.
```
```suggestion
```

sdelamo committed:
Update guides/micronaut-jpa-hibernate/micronaut-jpa-hibernate.adoc

sdelamo committed:
Update guides/micronaut-jpa-hibernate/micronaut-jpa-hibernate.adoc

timyates commented:
Something weird is going on here...

Adding

```
    annotationProcessor("io.micronaut.data:micronaut-data-document-processor")
```

fixes this...

I suspect https://github.com/micronaut-projects/micronaut-starter/pull/2173 may be wrong 🤔

sdelamo commented:
Closed via https://github.com/micronaut-projects/micronaut-guides/pull/1421 and https://github.com/micronaut-projects/micronaut-guides/pull/1422
